frontend/templates/product/index.tsx

```diff
@@ -129,8 +129,27 @@ export const getServerSideProps: GetServerSideProps<ProductTemplateProps> =
          context.res.setHeader('X-Robots-Tag', 'noindex, follow');
       }
 
+      const codeToDomain =
+         product.enabledDomains?.reduce((acc, { code, domain }) => {
+            acc[code] = domain;
+            return acc;
+         }, {} as Record<string, string>) ?? {};
+      const storesWithProductUrls = stores.map((store) => {
+         const domain =
+            store.code === DEFAULT_STORE_CODE
+               ? new URL(store.url).origin
+               : codeToDomain[store.code];
```

Comment:
We probably don't need to special case the default store code anymore, since all enabled store codes are included in the metafield.

```suggestion
         const domain = codeToDomain[store.code];
```